### DISCUSSION TITLE
Update existing Photo and keep metadata

### DISCUSSION THREAD

**Discussion by lookingdown** [photoprism/photoprism#1919] [Q&A] [answered]
If I need to update a photo (like rotating because horizon not horizontal OR whatever) I download the photo I need to edit, edit it and Upload it again to Photonix with same name as downloaded and present in Photonix.. 

I expected the existing original Photo in Photonix to be updated and whatever existing metadata stay the same..

But a new original photo is created instead.  It would be desirable to be able to download and edit photos and return them to photonix updated/edited.  If I have a photo in an album download edit it and upload again...  the photo in the album stays the same and a new one is created from scratch... ?

Am I missing something is there a setting available to allow update existing photos??

I am using and import folder where I put the original photos fro the camera (or icloud) photonix ( originals folder where the photo I want to edit is downloaded from.. )

Thanks

**Comment by graciousgrey** [comment]
Are you sure you're in the right repository? We're PhotoPrism, not Photonix ;)

**Reply by graciousgrey** [reply]
In case you mean PhotoPrism: Why are you editing a downloaded version instead of the original image from your originals directory?

If you edit the metadata of a photo from your originals directory, using another tool, PhotoPrism will read the updated metadata during the next index. You just need to make sure that the tool you are using to edit the file is updating the FileModification Date.

**Comment by lookingdown** [comment, marked as answer]
Of course PhotoPrism .. not photonix... which I tried out as well.. sorry about that typo.
I am referring to editing a photo in the originals folder.. as such find a photo you want to edit in an Album for instance.
Click to open the photo in the lightbox .. beside the edit icon there is a download icon download the photo.. Edit the photo  Save the photo after editing upload the photo with the same name.. I Expect the existing photo to be updated in photoprism BUT a new photo is created instead.. 
My question is how to upload an edited photo so that the existing photo with the same name is Replaced.. i.e. if the horizon needed correction (rotate) be able to do that and the upload the photo tp replace existing with a corrected one.. 
I now I can do that on the filesystem.. But I need to do it in photoprism I setup photoprism for my wife to manages a large photo collection and the logical way seems to be able to update existing photos by downloading them and uploading them again with the edits ..  i.e. overwrite the existing photo in originals folder.. 

Thanks :)

**Reply by graciousgrey** [reply]
You don't need to download/upload the photos using the PhotoPrism Web UI. This will result in two separate photos in your originals folder and the PhotoPrism UI.

What you need to do, to update a photo's metadata without creating additional files:
- On the photos [edit dialog](https://docs.photoprism.app/user-guide/organize/edit/#edit-files) you see the location and name of the photo in your originals folder
- In your filesystem or webdav go to this location. (You can access your original file directly where it is stored e.g. using the filesystem (Finder on Mac or Explorer on Windows) when it's stored locally on your computer or using [webdav](https://docs.photoprism.app/user-guide/sync/webdav/) when it's stored on a server. )
- Edit the original photo and save it with the same name
- In PhotoPrism start [indexing](https://docs.photoprism.app/user-guide/library/originals/) the folder (a normal rescan is sufficient)

**Reply by lookingdown** [reply]
Thanks for reply.. I know I can do that on the filesystem.. (webdav et al) But I need to do it in photoprism I setup photoprism for my wife to manages a large photo collection and the logical way seems to be able to update existing photos by downloading them and uploading them again with the edit done.

Browsing 50 000 photos in webdav to pick the right one ( looking first for filename in photodetails ) is very cumbersome and counter intuitive for a non technical person it seems.. There is a download action for each file in details->files  And in the lightbox..  it would seem logical to view the files when encounter one that needs editing download and edit it and upload it (update it) ..

Could be an update action in files.. button or link..  for a photographer with 50 000 plus photos to have to get out of photoprism open a shared folder in webdav  find among these 50 000 photos the corresponding one by name is not a light task or intuitive approach since all the data is available in photoprism perhaps just missing an update photo action that saves the photo to the corresponding file..  : ) ..

**Reply by lastzero** [reply]
Many other users are happy they can edit files using standard apps like Photoshop directly in the filesystem without manually uploading and downloading them with a web browser. We'll eventually implement an easy-to-use rotate functionality as part of the new photo viewer. Note photos can have more than one file and when you use the regular web upload there is no way to know to which photo the file belongs. If anything, it could be implemented in the files tab of the photo edit dialog.

**Reply by lookingdown** [reply]
I will probably for now if not possible in photoprism let her tag photos with "edit" and in a separate web app show  thumbnails with photos marked with edit and edit there directly with minipaint or download the file and upload after finished editing. (using external mysql with photoprism query the database for files tagged edit.. ) 

But it seems like a good idea to be able to update a photo directly in photo prism.. :)

**Reply by lookingdown** [reply]
If anything, it could be implemented in the files tab of the photo edit dialog. Thats what I thought .. will try to get my head around how to build a docker image from github sources so edited frontend files are included/compiled ... (to add edit and save functionality or just updatephoto).. : ) .. 
By the way the photo to edit is the corresponding original file in the originals folder.. where it is saved after upload/import for that path I query the DB and pick  files->file_name of the chosen record..  (with help of a few joins)
Photoprism is terrific anyway not used to docker so a bit of a learning curve before I will be able to do what I want..
Thanks
